teps/0089-spire-support.md

```diff
@@ -0,0 +1,184 @@
+
+Currently, Tekton Chains observes Tekton and waits for TaskRuns to complete.
+Once it sees that a TaskRun has completed it tries to sign any artifacts that were built and also generates provenance.
+
+There are a couple issues with this:
+1. Tekton Chains has no way to verify that a TaskRun in the cluster was created by Tekton
+2. Tekton Chains has no way to verify that the TaskRun it received wasn't modified by anybody other than Tekton
+
+This is where SPIRE comes in!
+SPIRE can be used to request signatures and certificates (SVIDs) for a given workload (this workload could be a result, or in our case an entire TaskRun yaml).
+SPIRE can be configured to only issue SVIDs to certain k8s workloads (e.g., you can set up SPIRE to only issue SVIDs to the Tekton controller) with a SPIRE [k8s Workload Attestor](https://github.com/spiffe/spire/blob/main/doc/plugin_agent_workloadattestor_k8s.md).
+This property is a critical feature, as it ensures that a TaskRun was created by Tekton.
+
+The basic flow would look like this:
+1. Tekton Pipelines receives a TaskRun config, and generates the Pod for it as usual
+1. Tekton Pipelines runs the TaskRun, and then requests a signature over the completed TaskRun yaml from SPIRE
+1. SPIRE has been set up to only issue signatures and SVIDs to the Tekton controller, which it does
+1. Tekton Pipelines stores the SVID and signature received from SPIRE as annotations on the TaskRun
+1. Tekton Chains verifies that SVID and signature against the TaskRun yaml
+1. If verification is successful, Chains proceeds normally. Otherwise, it stops and doesn't sign anything!
+
+With the SPIRE integration, we successfully mitigate the two issues mentioned above.
+Tekton Chains can be confident the TaskRun was created by Tekton, since only Tekton workloads can have SPIRE issued SVIDs.
+Tekton Chains can also verify that the TaskRun yaml wasn't modified by anybody other than the Tekton controller, because the signature exists over the entire TaskRun yaml.
+
+_NOTE: A cluster administrator could still potentially threaten this model by disabling SPIRE or changing settings on the k8s Workload Attestor. At SLSA 3, it is still allowed that some trusted administrators will have root access to the build system. This will eventually need to be addressed for SLSA 4_.
+
+
+## Motivation
+
+Security! =)
+
+We'll also need this feature for Tekton to achieve [SLSA 3](https://slsa.dev/levels), which requires Non-falsifiable Provenance.
+
+### Requirements
+* SPIRE is configured to only issue SVIDs to the Tekton controller via a [k8s Workload Attestor](https://github.com/spiffe/spire/blob/main/doc/plugin_agent_workloadattestor_k8s.md)
+
+### Goals
+
+* Once a TaskRun completes, Pipelines can request a signature over the TaskRun yaml from SPIRE
+* Pipelines can store the signature and SVID on the TaskRun itself as annotaitons
+* Chains should be able to verify a TaskRun wasn't modified after execution via the signature and SVID provided
```

Comment:
i am almost definitely overthinking this, but what if the TaskRun is modified during execution? it sounds like that's out of scope?